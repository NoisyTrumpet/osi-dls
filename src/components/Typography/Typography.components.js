import classNames from 'classnames';

/* TypographyDetails returns 2 props: 'html_tag' and 'className' */
const TypographyDetails = (defaultTag, className) => ({ defaultTag, className: classNames('typography', className) });

/*
TypographyComponents is a Key / Value pair
- Key: a valid type of <Typography />
- Value: a TypographyDetails object (defined above)
*/
const TypographyComponents = {
  'headline-1':
    TypographyDetails('h1', 'typography-headline typography-headline-1'),
  'headline-2':
    TypographyDetails('h2', 'typography-headline typography-headline-2'),
  'headline-3':
    TypographyDetails('h3', 'typography-headline typography-headline-3'),
  'headline-4':
    TypographyDetails('h4', 'typography-headline typography-headline-4'),
  'headline-5':
    TypographyDetails('h5', 'typography-headline typography-headline-5'),
  'headline-6':
    TypographyDetails('h6', 'typography-headline typography-headline-6'),
  'price-unit':
    TypographyDetails('span', 'typography-price-unit'),
  'price-unit-strikethrough':
    TypographyDetails('span', 'typography-price-unit-strikethrough'),
  'price-sale':
    TypographyDetails('span', 'typography-price-sale'),
  cta:
    TypographyDetails('span', 'typography-cta'),
  'cta-tertiary':
    TypographyDetails('span', 'typography-cta-tertiary'),
  subtitle:
    TypographyDetails('span', 'typography-subtitle'),
  'field-label':
    TypographyDetails('span', 'typography-field-label'),
  body:
    TypographyDetails('span', 'typography-body'),
  'body-medium':
    TypographyDetails('span', 'typography-body typography-body-medium'),
  eyebrow:
    TypographyDetails('span', 'typography-eyebrow'),
  'text-link':
    TypographyDetails('span', 'typography-text-link'),
  'input-text':
    TypographyDetails('span', 'typography-input-text'),
  caption:
    TypographyDetails('span', 'typography-caption'),
  helper:
    TypographyDetails('span', 'typography-helper'),
  breadcrumb:
    TypographyDetails('span', 'typography-breadcrumb'),
  srt:
    TypographyDetails('span', 'screen-reader-text'),
  licensing:
    TypographyDetails('span', 'typography-licensing'),
  subtext:
    TypographyDetails('span', 'typography-subtext'),
};

export default TypographyComponents;
