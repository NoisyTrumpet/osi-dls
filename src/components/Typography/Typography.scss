// Body
.typography-body {
  color: map-get($colors, 'gray5');
  font-size: rem(15px);
  letter-spacing: 0px;
  @include fontSans($gtAmerican, $gtAmerican, 'sans-serif', 84, 700);
}

.typography-breadcrumb,
.typography-body-medium,
.typography-cta-tertiary,
.typography-caption,
.typography-eyebrow,
.typography-helper,
.typography-input-text,
.typography-licensing,
.typography-subtext,
.typography-subtitle,
.typography-field-label,
.typography-text-link,
.typography-price-unit,
.typography-price-unit-strikethrough,
.typography-price-sale {
  @include fontSans($gtAmerican, $gtAmerican, 'sans-serif', 114, 700);
}

// Call To Action
.typography-cta {
  color: map-get($colors, 'primary');
  font-size: rem(14px);
  letter-spacing: rem(0.7px);
  text-transform: uppercase;
  @include fontSans($gtAmerican, $gtAmerican, 'sans-serif', 700, 700);
}

// Call To Action - Tertiary
.typography-cta-tertiary {
  color: map-get($colors, 'primary');
  font-size: rem(14px);
  letter-spacing: 0px;
}

//Caption
.typography-caption {
  color: map-get($colors, 'util-error');
  font-size: rem(13px);
  letter-spacing: 0px;
}

// Eyebrow
.typography-eyebrow {
  color: map-get($colors, 'gray5');
  font-size: rem(14px);
  text-transform: uppercase;
  letter-spacing: rem(0.8px);
}

// Headline
.typography-headline {
  color: map-get($colors, 'primary');
  @include fontSerif;
}

.typography-headline-1 {
  font-size: rem(64px);
  letter-spacing: rem(-1px);
}

.typography-headline-2 {
  font-size: rem(46px);
  letter-spacing: rem(-0.8px);
}

.typography-headline-3 {
  font-size: rem(36px);
  letter-spacing: rem(-0.5px);
}

.typography-headline-4 {
  font-size: rem(31px);
  letter-spacing: rem(-0.4px);
}

.typography-headline-5 {
  font-size: rem(24px);
  letter-spacing: rem(-0.2px);
}

.typography-headline-6 {
  font-size: rem(22px);
  letter-spacing: 0px;
}

// Price
.typography-price-unit {
  font-size: rem(22px);
}

.typography-price-unit-strikethrough {
  text-decoration: line-through;
  font-size: rem(16px);
}

.typography-price-sale {
  font-size: rem(22px);
}

// Helper
.typography-helper {
  color: map-get($colors, 'gray5');
  font-size: rem(14px);
  letter-spacing: 0px;
}

// Input Text
.typography-input-text {
  color: map-get($colors, 'gray5');
  font-size: rem(16px);
  letter-spacing: 0px;
}

// Subtitle
.typography-subtitle {
  color: map-get($colors, 'primary');
  font-size: rem(18px);
  letter-spacing: 0px;
}

// Field Label
.typography-field-label {
  color: map-get($colors, 'gray4');
  font-size: rem(15px);
  letter-spacing: rem(0.3px);
}

// Text Link
.typography-text-link {
  color: map-get($colors, 'accentB');
  font-size: rem(15px);
  letter-spacing: 0px;
  text-decoration: underline;
}

// Breadcrumb
.typography-breadcrumb,
.typography-licensing,
.typography-subtext {
  color: map-get($colors, 'gray5');
  font-size: rem(13px);
  letter-spacing: 0px;
}

// Screen Reader Text
.screen-reader-text {
  @extend %screen-reader-text;
}

// Colors
@each $name, $color in $colors {
  .color-#{$name},
  .color-#{$name}:before {
    color: $color;
  }
}
