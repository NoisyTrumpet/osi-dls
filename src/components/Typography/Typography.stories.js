import React from "react";
import { storiesOf } from "@storybook/react";
import { withSmartKnobs } from "storybook-addon-smart-knobs";
import { Typography } from "./";

const parameters = {
  component: Typography,
  componentSubtitle: "Utilize <Typography /> to present text",
  jest: ["Typography"]
};

storiesOf("components/Typography", module)
  .addParameters(parameters)
  .addDecorator(withSmartKnobs({ ignoreProps: [] }))
  .add("Default ", () => (
    <Typography variant="headline-1">Typography</Typography>
  ))
  .add("Text Align - Right", () => (
    <div style={{ textAlign: "right" }}>
      <Typography variant="headline-1">Typography</Typography>
    </div>
  ))
  .add("Text Align - Center", () => (
    <div style={{ textAlign: "center" }}>
      <Typography variant="headline-1">Typography</Typography>
    </div>
  ));
