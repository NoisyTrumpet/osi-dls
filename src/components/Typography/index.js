export { default as Typography } from './Typography'; // eslint-disable-line import/prefer-default-export
