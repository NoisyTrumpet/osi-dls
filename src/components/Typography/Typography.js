import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import TypographyComponents from './Typography.components';
import './Typography.scss';

/**
 * Utilize `Typography` to present text
 */
const Typography = ({ addClass, variant, htmlTagOverride, color, children, ...rest }) => {
  if (!(variant in TypographyComponents)) {
    return (null);
  }

  const {
    defaultTag,
    className,
  } = TypographyComponents[variant];

  const colorClass = {
    [`color-${color}`]: color,
  };

  const newProps = {
    className: classNames(className, colorClass, addClass),
    ...rest,
  };

  const tag = htmlTagOverride || defaultTag;
  return React.createElement(tag, newProps, children);
};

Typography.variants = Object.keys(TypographyComponents);

Typography.propTypes = {
  /** Variant determines the styling for the component */
  variant: PropTypes.oneOf([
    'headline-1',
    'headline-2',
    'headline-3',
    'headline-4',
    'headline-5',
    'headline-6',
    'price-unit',
    'price-unit-strikethrough',
    'price-sale',
    'cta',
    'cta-tertiary',
    'subtitle',
    'field-label',
    'body',
    'body-medium',
    'eyebrow',
    'text-link',
    'input-text',
    'caption',
    'helper',
    'breadcrumb',
    'srt',
    'licensing',
    'subtext',
  ]).isRequired,
  /** The content of the component */
  children: PropTypes.node,
  /** Overides the default tag that the component is rendered as */
  htmlTagOverride: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'p', 'span', 'address', 'div']),
  /** Color provides the component with a class name that affects the color of text */
  color: PropTypes.string,
  /** addClass appends a string to the component's className prop */
  addClass: PropTypes.string,
};

export default Typography;
