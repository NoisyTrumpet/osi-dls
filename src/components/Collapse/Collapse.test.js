import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Collapse } from './';
import { Typography } from '../Typography';

const content = (
  <>
    <Typography variant="body" htmlTagOverride="p">
      Personalize a handwritten Hallmark Card and we&apos;ll mail it for you!
    </Typography>
    <Typography variant="body" htmlTagOverride="p">
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
      incididunt ut labore et dolore magna aliqua. Habitasse platea dictumst quisque
      sagittis purus sit amet volutpat consequat. At augue eget arcu dictum. Hac habitasse
      platea dictumst quisque sagittis purus sit. Elementum pulvinar etiam non quam lacus
      suspendisse faucibus. Dictumst quisque sagittis purus sit amet volutpat consequat
      mauris. A condimentum vitae sapien pellentesque habitant morbi tristique senectus.
      Eget est lorem ipsum dolor sit. Ut consequat semper viverra nam libero justo
      laoreet sit amet. Netus et malesuada fames ac turpis egestas sed. Congue quisque
      egestas diam in arcu cursus euismod quis viverra. Lacus sed turpis tincidunt
      id aliquet risus feugiat in ante. Aliquam vestibulum morbi blandit cursus risus
      at ultrices mi. Diam maecenas sed enim ut sem viverra aliquet. Leo urna molestie
      at elementum. Pretium fusce id velit ut tortor pretium viverra.
    </Typography>
    <Typography variant="body" htmlTagOverride="p">
      Mauris in aliquam sem fringilla ut morbi. Morbi leo urna molestie at elementum eu.
      Faucibus interdum posuere lorem ipsum dolor sit amet consectetur adipiscing. Nulla
      at volutpat diam ut venenatis tellus in metus. Vestibulum mattis ullamcorper velit
      sed ullamcorper morbi. Leo urna molestie at elementum eu facilisis. Ut faucibus
      pulvinar elementum integer. Vitae ultricies leo integer malesuada. Accumsan sit
      amet nulla facilisi. Nascetur ridiculus mus mauris vitae ultricies leo integer.
      Mauris vitae ultricies leo integer malesuada.
    </Typography>
    <Typography variant="body" htmlTagOverride="p">
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
      incididunt ut labore et dolore magna aliqua. Habitasse platea dictumst quisque
      sagittis purus sit amet volutpat consequat. At augue eget arcu dictum. Hac habitasse
      platea dictumst quisque sagittis purus sit. Elementum pulvinar etiam non quam lacus
      suspendisse faucibus. Dictumst quisque sagittis purus sit amet volutpat consequat
      mauris. A condimentum vitae sapien pellentesque habitant morbi tristique senectus.
      Eget est lorem ipsum dolor sit. Ut consequat semper viverra nam libero justo
      laoreet sit amet. Netus et malesuada fames ac turpis egestas sed. Congue quisque
      egestas diam in arcu cursus euismod quis viverra. Lacus sed turpis tincidunt
      id aliquet risus feugiat in ante. Aliquam vestibulum morbi blandit cursus risus
      at ultrices mi. Diam maecenas sed enim ut sem viverra aliquet. Leo urna molestie
      at elementum. Pretium fusce id velit ut tortor pretium viverra.
    </Typography>
  </>
);

describe('Collapse', () => {
  test('Should render', () => {
    const wrapper = shallow(<Collapse>{content}</Collapse>);
    expect(wrapper.length).toBe(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
