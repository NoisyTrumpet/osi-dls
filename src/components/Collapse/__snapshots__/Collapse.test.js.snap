// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse Should render 1`] = `
<div
  className="collapse-container default collapsed"
>
  <div
    className="collapse-content"
  >
    <Typography
      htmlTagOverride="p"
      variant="body"
    >
      Personalize a handwritten Hallmark Card and we'll mail it for you!
    </Typography>
    <Typography
      htmlTagOverride="p"
      variant="body"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Habitasse platea dictumst quisque sagittis purus sit amet volutpat consequat. At augue eget arcu dictum. Hac habitasse platea dictumst quisque sagittis purus sit. Elementum pulvinar etiam non quam lacus suspendisse faucibus. Dictumst quisque sagittis purus sit amet volutpat consequat mauris. A condimentum vitae sapien pellentesque habitant morbi tristique senectus. Eget est lorem ipsum dolor sit. Ut consequat semper viverra nam libero justo laoreet sit amet. Netus et malesuada fames ac turpis egestas sed. Congue quisque egestas diam in arcu cursus euismod quis viverra. Lacus sed turpis tincidunt id aliquet risus feugiat in ante. Aliquam vestibulum morbi blandit cursus risus at ultrices mi. Diam maecenas sed enim ut sem viverra aliquet. Leo urna molestie at elementum. Pretium fusce id velit ut tortor pretium viverra.
    </Typography>
    <Typography
      htmlTagOverride="p"
      variant="body"
    >
      Mauris in aliquam sem fringilla ut morbi. Morbi leo urna molestie at elementum eu. Faucibus interdum posuere lorem ipsum dolor sit amet consectetur adipiscing. Nulla at volutpat diam ut venenatis tellus in metus. Vestibulum mattis ullamcorper velit sed ullamcorper morbi. Leo urna molestie at elementum eu facilisis. Ut faucibus pulvinar elementum integer. Vitae ultricies leo integer malesuada. Accumsan sit amet nulla facilisi. Nascetur ridiculus mus mauris vitae ultricies leo integer. Mauris vitae ultricies leo integer malesuada.
    </Typography>
    <Typography
      htmlTagOverride="p"
      variant="body"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Habitasse platea dictumst quisque sagittis purus sit amet volutpat consequat. At augue eget arcu dictum. Hac habitasse platea dictumst quisque sagittis purus sit. Elementum pulvinar etiam non quam lacus suspendisse faucibus. Dictumst quisque sagittis purus sit amet volutpat consequat mauris. A condimentum vitae sapien pellentesque habitant morbi tristique senectus. Eget est lorem ipsum dolor sit. Ut consequat semper viverra nam libero justo laoreet sit amet. Netus et malesuada fames ac turpis egestas sed. Congue quisque egestas diam in arcu cursus euismod quis viverra. Lacus sed turpis tincidunt id aliquet risus feugiat in ante. Aliquam vestibulum morbi blandit cursus risus at ultrices mi. Diam maecenas sed enim ut sem viverra aliquet. Leo urna molestie at elementum. Pretium fusce id velit ut tortor pretium viverra.
    </Typography>
  </div>
</div>
`;
